tests/phpunit/tests/test-olt-manager.php

```diff
@@ -21,4 +30,45 @@ public function test_polylang_first() {
 		delete_option( 'active_plugins' );
 		delete_option( 'active_sitewide_plugins' );
 	}
+
+	public function test_load_textdomains() {
+		// Clean up!
+		unset( $GLOBALS['wp_themes'], $GLOBALS['l10n'], $GLOBALS['l10n_unloaded'] );
+
+		/** @var WP_Textdomain_Registry $wp_textdomain_registry */
+		global $wp_textdomain_registry;
+
+		$wp_textdomain_registry = new WP_Textdomain_Registry();
+
+		update_option( 'WPLANG', '' );
+
+		// Copy language file.
+		@mkdir( DIR_TESTDATA );
+		@mkdir( WP_LANG_DIR );
+		@mkdir( WP_LANG_DIR . '/plugins' );
+		copy( PLL_TEST_DATA_DIR . 'fr_FR.mo', WP_LANG_DIR . '/plugins/foo-fr_FR.mo' );
+
+		new PLL_OLT_Manager();
+		$options                 = PLL_Install::get_default_options();
+		$options['force_lang']   = 0;
+		$options['default_lang'] = 'en';
+		$model                   = new PLL_Model( $options );
+		$links_model             = $model->get_links_model();
+		$frontend                = new PLL_Frontend( $links_model );
+
+		/*
+		 *  Calls `_load_textdomain_just_in_time()` *before* the current language is defined!
+		 */
+		__( 'Dashboard', 'foo' ); // phpcs:ignore WordPress.WP.I18n.TextDomainMismatch
```

Comment:
I suggest to perform an assertion here to check that no translation is loaded.
```suggestion
		$this->assertSame( 'Dashboard', __( 'Dashboard', 'foo' ) ); // phpcs:ignore WordPress.WP.I18n.TextDomainMismatch
```

In fact to be even more complete we could test with a WPLANG option set to a 3rd language (not en_US or fr_FR), let say de_DE, to make sure that foo-de_DE.mo is not loaded. What do you think?

Reply:
Good idea, done in 6cb66abb.